compiler/rustc_metadata/src/rmeta/decoder.rs

```diff
@@ -1071,15 +1071,30 @@ impl<'a, 'tcx> CrateMetadataRef<'a> {
         }
     }
 
-    /// Iterates over each child of the given item.
-    fn each_child_of_item(&self, id: DefIndex, mut callback: impl FnMut(Export), sess: &Session) {
+    fn get_associated_item_def_ids(
+        &'a self,
+        id: DefIndex,
+        sess: &'a Session,
+    ) -> impl Iterator<Item = DefId> + 'a {
+        self.root
+            .tables
+            .children
+            .get(self, id)
+            .unwrap_or_else(Lazy::empty)
+            .decode((self, sess))
+            .map(|child_index| self.local_def_id(child_index))
+    }
+
+    /// Iterates over all named children of the given module,
+    /// including both proper items and reexports.
+    /// Module here is understood in name resolution sense - it can be a `mod` item,
+    /// or a crate root, or an enum, or a trait.
+    fn get_module_children(&self, id: DefIndex, mut callback: impl FnMut(Export), sess: &Session) {
```

Comment:
I like the previous name more. Maybe something like `each_module_child`? `get*` feels like it should be returning something.

Reply:
This is the only function in this module that uses a callback instead of returning something.
I want to change it to returning an iterator, it just needs to be simplified before that.